samples/net/udp/src/main.c

```diff
@@ -0,0 +1,202 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <stdio.h>
+#include <zephyr/kernel.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/logging/log_ctrl.h>
+#include <zephyr/sys/reboot.h>
+#include <zephyr/net/socket.h>
+#include <zephyr/net/conn_mgr_connectivity.h>
+#include <zephyr/net/conn_mgr_monitor.h>
+
+LOG_MODULE_REGISTER(udp_sample, CONFIG_UDP_SAMPLE_LOG_LEVEL);
+
+#define UDP_IP_HEADER_SIZE 28
+
+/* Macros used to subscribe to specific Zephyr NET management events. */
+#define L4_EVENT_MASK (NET_EVENT_L4_CONNECTED | NET_EVENT_L4_DISCONNECTED)
+#define CONN_LAYER_EVENT_MASK (NET_EVENT_CONN_IF_FATAL_ERROR)
+
+/* Macro called upon a fatal error, reboots the device. */
+#define FATAL_ERROR()					\
+	LOG_ERR("Fatal error! Rebooting the device.");	\
+	LOG_PANIC();					\
+	IF_ENABLED(CONFIG_REBOOT, (sys_reboot(0)))
+
+/* Zephyr NET management event callback structures. */
+static struct net_mgmt_event_callback l4_cb;
+static struct net_mgmt_event_callback conn_cb;
+
+/* Variables used to performs socket operations. */
+static int client_fd;
+static struct sockaddr_storage host_addr;
+
+/* Forward declarations */
+static void server_transmission_work_fn(struct k_work *work);
+
+/* Work item used to schedule periodic UDP transmissions. */
+static K_WORK_DELAYABLE_DEFINE(server_transmission_work, server_transmission_work_fn);
+
+static void server_transmission_work_fn(struct k_work *work)
+{
+	int err;
+	char buffer[CONFIG_UDP_SAMPLE_DATA_UPLOAD_SIZE_BYTES] = {"\0"};
+
+	LOG_INF("Transmitting UDP/IP payload of %d bytes to the "
+		"IP address %s, port number %d",
+		CONFIG_UDP_SAMPLE_DATA_UPLOAD_SIZE_BYTES + UDP_IP_HEADER_SIZE,
+		CONFIG_UDP_SAMPLE_SERVER_ADDRESS_STATIC,
+		CONFIG_UDP_SAMPLE_SERVER_PORT);
+
+	err = send(client_fd, buffer, sizeof(buffer), 0);
+	if (err < 0) {
+		LOG_ERR("Failed to transmit UDP packet, %d", errno);
+		return;
+	}
+
+	(void)k_work_reschedule(&server_transmission_work,
+				K_SECONDS(CONFIG_UDP_SAMPLE_DATA_UPLOAD_FREQUENCY_SECONDS));
+}
+
+static int server_init(void)
+{
+	struct sockaddr_in *server4 = ((struct sockaddr_in *)&host_addr);
+
+	server4->sin_family = AF_INET;
+	server4->sin_port = htons(CONFIG_UDP_SAMPLE_SERVER_PORT);
+
+	inet_pton(AF_INET, CONFIG_UDP_SAMPLE_SERVER_ADDRESS_STATIC, &server4->sin_addr);
+
+	return 0;
+}
+
+static void server_disconnect(void)
+{
+	(void)close(client_fd);
+}
+
+static int server_connect(void)
```

Comment:
I'd personally use `client_start/stop()` instead of `server_connect/disconnect()`, we're not really establishing any connection with UDP. But I don't mind much if you want to keep it as is.

Reply:
Ill fix this